packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -229,9 +216,10 @@ class Observatory {
   }
 
   Future<Response> flutterExit(String isolateId) {
-    return peer.sendRequest('ext.flutter.exit', <String, dynamic>{
-      'isolateId': isolateId
-    }).then((dynamic result) => new Response(result));
+    return peer
+      .sendRequest('ext.flutter.exit', <String, dynamic>{ 'isolateId': isolateId })
+      .then((dynamic result) => new Response(result))
+      .timeout(new Duration(seconds: 2), onTimeout: () => null);
```

Comment:
Can't tell if it's important here, but this timeout will not cancel the underlying HTTP request, which may result in undefined behavior. Here are some options for propagating cancellation all the way to the request itself:

- Use `Stream` API instead of `Future`. Stream subscriptions are chained (if intermediate subscriptions are implemented correctly) such that when the consumer subscription is cancelled the cancellation event propagates to the root subscription, giving you a chance to clean-up resources.
- There are extensions available that implement cancelable futures: [CancelableCompleter (package:async)](https://www.dartdocs.org/documentation/async/1.11.0/async/CancelableCompleter-class.html), [CencelableFuture (package:analyzer)](https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/src/cancelable_future.dart#L163)